### DISCUSSION TITLE
Vector Index Construction Documentation

### DISCUSSION THREAD

**Discussion by mykittylover** [milvus-io/milvus#35787] [Q&A and General discussion]
I noticed that vector index construction in C++ is not supported. Could you tell me where the documentation for vector index construction is? I would like to see how scalar + vector indexing is constructed. Thank you!

**Comment by yhmo** [comment]
We don't have technical documentation about the details of the source code. Since it is oper source, you can read the source code from the entry.
When milvus receives a crate_index request, the index node will get an index task to build index. The indexcgowrapper.CreateIndex() is called in the index task:
https://github.com/milvus-io/milvus/blob/73be0ba9415192130b013fa118cd31dac63ca751/internal/indexnode/task_index.go#L300C18-L300C33

milvus::indexbuilder::IndexFactory::GetInstance().CreateIndex() is called:
https://github.com/milvus-io/milvus/blob/b3089b5bdc7f3a1699c6674b8c7d72c5a4562be4/internal/core/src/indexbuilder/index_c.cpp#L213

Here we can see different index builder is created according to index type:
https://github.com/milvus-io/milvus/blob/master/internal/core/src/indexbuilder/IndexFactory.h#L52
https://github.com/milvus-io/milvus/blob/master/internal/core/src/index/IndexFactory.h

For scalar index, InvertedIndexTantivy/BitmapIndex/HybridScalarIndex/StringIndexMarisa, etc. All the scalar index classes are implemented here:
https://github.com/milvus-io/milvus/tree/master/internal/core/src/index

For vector index, we can see the VectorMemIndex and VectorDiskAnnIndex have a member: knowhere::Index<knowhere::IndexNode> index_;
https://github.com/milvus-io/milvus/blob/70bc0e45387eb2594f35d970845a99c638631e93/internal/core/src/index/VectorMemIndex.h#L99
https://github.com/milvus-io/milvus/blob/70bc0e45387eb2594f35d970845a99c638631e93/internal/core/src/index/VectorDiskIndex.h#L101

knowhere::IndexNode is implemented in the knowhere repo:
https://github.com/zilliztech/knowhere/blob/main/include/knowhere/index/index_node.h
https://github.com/zilliztech/knowhere/tree/main/src/index
